Customisable Create Label

duncanmcclean opened the pull request:
This pull request introduces the ability to customise the 'Create Entry' text displayed various places in the Control Panel. Closes #3569.

* [X] Collection index
* [X] Collection create page title
* [x] Collection widget

jasonvarga commented:
I wonder if there's a simpler way to do it.

Like, if we just have a convention where if you have a `{handle}_collection_create_entry` `blog_collection_create_entry` translation, that it'll use that.

Having a `createLabel` method is fine, but maybe it's just a getter that looks for that translation.

```php
public function createLabel()
{
  $key = $this->handle().'_collection_create_entry';
  $translation = __($key);

  if ($translation === $key) {
    return __('Create Entry');
  }

 return $translation;
}
```

I know the next request will be to translate the button text, which this PR doesn't support in its current state.

duncanmcclean commented:
That's a good idea, didn't think about translations. I'll mark this pull request as a draft until I update this PR 😄

duncanmcclean commented:
Updated this PR to use translations instead. I also simplified the test to just get the default value. I wasn't sure if there's a way to mock a translation during a test or not.
